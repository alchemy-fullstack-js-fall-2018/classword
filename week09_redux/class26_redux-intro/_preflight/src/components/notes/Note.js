import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

const Note = ({ note }) => {
  if(!note) return <Redirect to={ROUTES.HOME.linkTo()} />;

  const { title, body } = note;
  return (
    <div>
      <h2>{title}</h2>
      <p>{body}</p>
    </div>
  );
};

Note.propTypes = {
  note: PropTypes.object.isRequired
};

export default Note;
