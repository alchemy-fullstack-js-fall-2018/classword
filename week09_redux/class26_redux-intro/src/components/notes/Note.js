import React from 'react';
import PropTypes from 'prop-types';

const Note = ({ note }) => {
  const { title, body } = note;
  return (
    <div>
      <h3>{title}</h3>
      <p>{body}</p>
    </div>
  );
};

Note.propTypes = {
  note: PropTypes.object.isRequired
};

export default Note;
