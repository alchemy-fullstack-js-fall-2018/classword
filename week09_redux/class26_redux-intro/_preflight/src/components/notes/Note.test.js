import React from 'react';
import { shallow } from 'enzyme';
import Note from './Note';

jest.mock('../../routes/routes.js');

describe('Note', () => {
  const note = { title: 'Hi There', body: 'You there' };
  const wrapper = shallow(<Note note={note} />);

  it('renders a Note component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a note title', () => {
    expect(wrapper.html()).toContain(note.title);
  });

  it('displays a note body', () => {
    expect(wrapper.html()).toContain(note.body);
  });
});
