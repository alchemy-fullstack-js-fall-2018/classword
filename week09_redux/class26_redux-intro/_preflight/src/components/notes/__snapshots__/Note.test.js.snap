// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders a Note component 1`] = `
<div>
  <h2>
    Hi There
  </h2>
  <p>
    You there
  </p>
</div>
`;
